src/modules/UI.js

```diff
@@ -1,23 +1,204 @@
+    const checkboxValue = checkbox.value;
+    const listItem = checkbox.parentNode;
+    const checkboxItself = listItem.querySelector('.checkbox');
+    const taskDescription = listItem.querySelector('.task-description');
+    const blueTick = listItem.querySelector('.blue-tick');
+
+    if (taskDescription) {
+      if (checkbox.checked) {
+        const completedTask = {
+          description: checkboxValue,
+          completed: true,
+          index: checkboxID,
+        };
+        UI.completedTasks.push(completedTask);
+        taskDescription.classList.add('completed');
+        checkboxItself.classList.add('hide');
+        blueTick.classList.remove('hide');
+        checkbox.dataset.completedTask = JSON.stringify(completedTask);
+      } else { taskDescription.classList.remove('completed'); }
+    }
+  };
+
+  // edit task
+  static editTask = (taskOption) => {
+    const taskID = parseInt(taskOption.id, 10);
+    const listItem = taskOption.closest('.to-do-list-item');
+    const editInput = listItem.querySelector('.edit-input');
+    const checkbox = listItem.querySelector('.checkbox');
+    const taskDescription = listItem.querySelector('.task-description');
+    const menuIcon = listItem.querySelector('.menu');
+    const trashBin = listItem.querySelector('.trash-task-btn');
+
+    if (listItem) {
+      const task = UI.toDoList.findIndex((t) => t.index === taskID);
+      if (task !== -1) {
+        // Check if task exists
+        listItem.classList.add('edit-bg-color');
+        editInput.classList.remove('hide');
+        trashBin.classList.remove('hide');
+        checkbox.classList.add('hide');
+        menuIcon.classList.add('hide');
+        taskDescription.classList.add('hide');
+        trashBin.disabled = true;
+        editInput.addEventListener('input', (e) => UI.deleteTask(e, taskID, editInput, trashBin));
+        editInput.addEventListener('keypress', (e) => UI.updateTask(e, taskID, editInput));
+      }
+    }
+  };
+
+  // enable delete if fields are empty
+  static updateTask(e, taskID, editInput) {
+    const inputValue = editInput.value;
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      if (inputValue !== '') {
+        const taskIndex = UI.toDoList.findIndex(
+          (task) => task.index === taskID,
+        );
+        if (taskIndex !== -1) {
+          UI.toDoList[taskIndex].description = inputValue;
+          Store.updateTasks(UI.toDoList);
+          UI.toDoList = Store.getTasks();
+          UI.displayToDoList();
+        }
+      }
+    }
+  }
+
+  // enable delete if fields are empty
+  static deleteTask(e, taskID, editInput, trashBin) {
+    e.preventDefault();
+    const inputValue = editInput.value;
+    if (inputValue === '') {
+      trashBin.disabled = false;
+      trashBin.addEventListener('click', (e) => {
+        Store.removeTask(e, taskID);
+        UI.toDoList = Store.getTasks();
+        UI.reOrderList();
+        UI.displayToDoList();
+      });
+    }
+  }
```

Comment:
- Good job implementing all the required funtionalities. However, when I click on the trash button, the task is not being deleted neither from the List, nor from the LocalStorage. Kindly fix that according to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20deleting%20a%20task%20(remove%20an%20element%20from%20the%20array).)

![Screenshot from 2023-05-12 14-58-08](https://github.com/DebyGrey/ToDo-list/assets/97834160/2c3772b7-9d91-4ea4-be5f-136c33619e53)

Reply:
@GedeonTS  it works actually just that my implementation is different. I thought I saw in the video that the button becomes active only when the user has cleared the input. I will correct it now, thanks.